### DISCUSSION TITLE
What should we use for naming functions/variables/etc.?

### DISCUSSION THREAD

**Discussion by Panquesito7** [TheAlgorithms/C-Plus-Plus#2488] [Polls]
I didn't notice early (thanks to @realstealthninja for spotting this), but we don't have a rule/convention of what should we use for variables and functions: PascalCase, camelCase, or snake_case.
What do you think we should use? Feel free to vote in the poll below. 🙂

**Comment by vil02** [comment]
What about template parameters and macros?

I would stick to what is suggested in [Cpp best practices](https://github.com/cpp-best-practices/cppbestpractices/blob/master/03-Style.md#common-c-naming-conventions). My personal preference leans towards the convention used in the standard library and BOOST (i.e. the _second style_ from the link above). Have a look at [C++ Core Guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#nl10-prefer-underscore_style-names).

**Comment by KhanN-byte** [comment]
Hello, I just came across this repo and I really like it, it has tons of useful info. How active are the code reviewers for this github repo? I just see a bunch of open requests still outstanding. I just submitted one CR for a minor fix/tweak in the linked_list.cpp. Thanks!!
